components/table/table-cell.tsx

```diff
@@ -20,8 +21,11 @@ export type cellData = {
 }
 
 const TableCell: React.FC<Props> = ({ columns, row, rowIndex, emptyText, onCellClick }) => {
-  const { removeRow } = useTableContext()
+  const { removeRow, updateRow } = useTableContext()
   const actions: cellActions = {
+    update: function <T>(data: T) {
```

Comment:
(I forgot to submit the comments here. I just saw them just now)

Reply:
yeah i look into typescript docs and the infer but couldn't find how to "force" it. vscode is expecting a type for `data` and i don't know how to tell him infer from the args that will be passed by the user if that makes sense. happy to see you have a go at it when you have time because i'm not sure how to remove vscode typescript error and get the inference here.

I'll keep looking into it and will tell you if i find something in the meantime :)

Reply:
at first i wanted to say `data` of update was a Partial of Table's `data` but Table's data is typed as `any[]` so can't really infer type from it either i think.